File: native/mixdog-graph/tests/fixtures/parity/java/Sample.java
// Parity fixture: declaration and import shapes the graph reports for Java.
package com.acme.sample;

import java.util.List;
import java.util.concurrent.*;
import static java.util.Collections.emptyList;

public class Sample {
  private final String name;

  public Sample(String name) {
    this.name = name;
  }

  public String read(String key) {
    return name + key;
  }

  static List<String> empty() {
    return emptyList();
  }

  interface Listener {
    void onEvent(String event);
  }

  enum Mode {
    FAST,
    SLOW;

    boolean isFast() {
      return this == FAST;
    }
  }

  record Pair(String left, String right) {
    String joined() {
      return left + right;
    }
  }

  static class Inner {
    Callable<String> task() {
      return () -> "inner";
    }
  }
}

interface TopLevelListener {
  void handle();
}

enum TopLevelMode {
  ON,
  OFF
}

File: native/mixdog-graph/tests/fixtures/parity/kotlin/Sample.kt
// Parity fixture: declaration and import shapes the graph reports for Kotlin.
package com.acme.sample

import kotlin.math.max
import java.util.concurrent.TimeUnit as Unit

class Store(val name: String) {
    private var size: Int = 0

    fun read(key: String): String {
        fun nested(value: String) = value.trim()
        return nested("$name/$key")
    }

    companion object {
        fun create(name: String): Store = Store(name)
    }

    class Inner {
        fun ping(): Long = Unit.SECONDS.toMillis(1)
    }
}

interface Listener {
    fun onEvent(event: String)
}

enum class Mode {
    FAST,
    SLOW
}

object Registry {
    val all = mutableListOf<Store>()

    fun register(store: Store) {
        all.add(store)
    }
}

fun topLevel(values: List<Int>): Int = values.fold(0) { acc, value -> max(acc, value) }
